src/EditorFeatures/Test2/Classification/ClassificationTests.vb

```diff
@@ -22,8 +22,58 @@ Imports Microsoft.VisualStudio.Text.Tagging
 Imports Roslyn.Utilities
 
 Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Classification
-    <[UseExportProvider]>
+    <UseExportProvider>
     Public Class ClassificationTests
+        <Fact, WorkItem(66245, "https://github.com/dotnet/roslyn/pull/66245")>
+        Public Async Function TestClassificationAndHighlight1() As Task
+            Using workspace = TestWorkspace.Create(
+                <Workspace>
+                    <Project Language="C#" AssemblyName="TestAssembly" CommonReferences="true">
+                        <Document>
+                        using System.Text.RegularExpressions;
+
+                        class C
+                        {
+                           [| Regex |]re = new Regex("()");
+                        }
+                        </Document>
+                    </Project>
+                </Workspace>)
+
+                Dim document = workspace.CurrentSolution.Projects.Single().Documents.Single()
+                Dim text = Await document.GetTextAsync()
+                Dim referenceSpan = workspace.Documents.Single().SelectedSpans.Single()
+
+                Dim spansAndHighlightSpan = Await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(
+                    New DocumentSpan(document, referenceSpan),
+                    ClassificationOptions.Default, CancellationToken.None)
+
+                ' This is the classification of the line, starting at the beginning of the highlight, and going to the end of that line.
+                Assert.Equal(
+"(text, '<spaces>', [154..155))
+(class name, 'Regex', [155..160))
+(text, '<spaces>', [160..161))
```

Comment:
previously, the result classifications would start at 'Regex', despite the reference span starting before it.  This would cause an OOB when we tried to grab hte reference span out of the classifications.